numpyro/distributions/kl.py

```diff
@@ -134,6 +136,66 @@ def kl_divergence(p, q):
     return 0.5 * (var_ratio + t1 - 1 - jnp.log(var_ratio))
 
 
+@dispatch(MultivariateNormal, MultivariateNormal)
+def kl_divergence(p: MultivariateNormal, q: MultivariateNormal):
+    # cf https://statproofbook.github.io/P/mvn-kl.html
+
+    if p.event_shape != q.event_shape:
+        raise ValueError(
+            "Distributions must be have the same event shape, but are"
+            f" {p.event_shape} and {q.event_shape} for p and q, respectively."
+        )
+
+    if p.batch_shape != q.batch_shape:
+        raise ValueError(
+            "Distributions must be have the same batch shape, but are"
+            f" {p.batch_shape} and {q.batch_shape} for p and q, respectively."
+        )
+
+    assert len(p.event_shape) == 1, "event_shape must be one-dimensional"
+    D = p.event_shape[0]
+
+    assert p.mean.shape == p.batch_shape + p.event_shape
+    assert q.mean.shape == p.mean.shape
+
+    def _single_mvn_kl(p_mean, p_scale_tril, q_mean, q_scale_tril):
+        assert jnp.ndim(p_mean) == 1
+        assert jnp.ndim(q_mean) == 1
+        assert jnp.ndim(p_scale_tril) == 2
+        assert jnp.ndim(q_scale_tril) == 2
+
+        p_half_log_det = jnp.log(
+            jnp.diagonal(p_scale_tril)
+        ).sum(-1)
+        q_half_log_det = jnp.log(
+            jnp.diagonal(q_scale_tril)
+        ).sum(-1)
+        log_det_ratio = 2 * (p_half_log_det - q_half_log_det)
+
+        Lq_inv = solve_triangular(q_scale_tril, jnp.eye(D), lower=True)
+
+        tr = jnp.sum(jnp.diagonal(
+            Lq_inv.T @ (Lq_inv @ p_scale_tril) @ p_scale_tril.T
+        ))
+
+        z = jnp.matmul(Lq_inv, (p_mean - q_mean))
+        t1 = jnp.dot(z, z)
```

Comment:
You can use [_batch_mahalanobis](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/distributions/continuous.py#L1370) for this:
```
t1 = _batch_mahalanobis(q.scale_tril, q.loc - p.loc)
```